classes/controller/FrontController.php

```diff
@@ -445,6 +445,10 @@ public function getSeoFields()
                 break;
 
         }
+        // append page number
+        if ($p = Tools::getIntValue('p')) {
+            $canonical .= "?p=$p";
```

Comment:
This is not the right way to modify url.

At this point, we don't know what `$canonical` looks like. It could be `https://www.domain.com/en/rewrite/url`
but it could very easily be `https://www.domain.con/en/rewrite?id_category=10`

We can't simply concat `'?p=1` here